src/modules/complianceengine/src/lib/ComplianceEngineInterface.cpp

```diff
@@ -46,7 +48,23 @@ MMI_HANDLE ComplianceEngineMmiOpen(const char* clientName, const unsigned int ma
         OsConfigLogError(g_log, "ComplianceEngineMmiOpen(%s, %u): failed to create context", clientName, maxPayloadSizeBytes);
         return nullptr;
     }
-    auto* result = reinterpret_cast<void*>(new Engine(std::move(context)));
+    std::unique_ptr<ComplianceEngine::PayloadFormatter> formatter;
+    if (!strcmp(clientName, cModuleTestClientName))
+    {
+        OsConfigLogInfo(g_log, "ComplianceEngineMmiOpen(%s) using DebugFormatter", clientName);
+        formatter = std::unique_ptr<ComplianceEngine::PayloadFormatter>(new ComplianceEngine::DebugFormatter());
+    }
+    else if (!strcmp(clientName, cNRPClientName))
```

Comment:
I don't see the constant value of `cNRPClientName` being used anywhere - are we sure this is being correctly selected?